### ISSUE TITLE
Cant install using npm - broken dependency tree (@nivo/pie v0.67.0)

### ISSUE THREAD

**Issue by adavies1** [plouc/nivo#1462]
Good evening,

When trying to install the nivo/pie package by running `npm i @nivo/pie`, npm fails with the following error:

```
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! Found: react-dom@17.0.2
npm ERR! node_modules/react-spring/node_modules/react-dom
npm ERR!   peer react-dom@">=17.0" from react-three-fiber@5.3.22
npm ERR!   node_modules/react-spring/node_modules/react-three-fiber
npm ERR!     peer react-three-fiber@">=4.0" from @react-spring/three@9.0.0-rc.3
npm ERR!     node_modules/react-spring/node_modules/@react-spring/three
npm ERR!       @react-spring/three@"9.0.0-rc.3" from react-spring@9.0.0-rc.3
npm ERR!       node_modules/react-spring
npm ERR!         react-spring@"9.0.0-rc.3" from @nivo/core@0.67.0
npm ERR!         node_modules/@nivo/core
npm ERR!         1 more (@nivo/tooltip)
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer react-dom@"16.8.x" from react-konva@16.8.6
npm ERR! node_modules/react-spring/node_modules/react-konva
npm ERR!   peer react-konva@">=16.8" from @react-spring/konva@9.0.0-rc.3
npm ERR!   node_modules/react-spring/node_modules/@react-spring/konva
npm ERR!     @react-spring/konva@"9.0.0-rc.3" from react-spring@9.0.0-rc.3
npm ERR!     node_modules/react-spring
npm ERR!       react-spring@"9.0.0-rc.3" from @nivo/core@0.67.0
npm ERR!       node_modules/@nivo/core
npm ERR!       1 more (@nivo/tooltip)
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
```

**To Reproduce**
1. Create a blank folder and cd into it
2. Run `npm init` to quickly create a new package.json
3. Run `npm i @nivo/pie`

**Expected behavior**
Should install successfully

**Additional context**
I am running on Linux (Ubuntu), using nodeJS v15.12.0 and npm v7.7.6.
It seems your dependencies cannot agree on which version of `react-dom` to use.

Thank you,
Alex

**Comment by akigugale** [comment]
Having the same problem but with `@nivo/core`, @adavies1 could you find a solution to this?

**System Info**
I am running on MacOS (Big Sur 11.2.3), using nodeJS v14.0.7.0 and npm v7.8.0.

**Comment by akigugale** [comment]
Update: the package installs successfully on **npm v6**

**Comment by wyze** [comment]
This seems like an issue from a 3rd party dependency, react-spring. We've updated to the latest version for the next release, so this might be fixed. Otherwise I suggest using npm 6 or add the --legacy-peer-deps flag.

**Comment by blimmer** [comment]
Hey @wyze - I saw that there have been a few releases since this issue was closed, but we're still experiencing the same issue. Is there another issue for upgrading react-spring that we can subscribe to to get updated on when we can upgrade to NPM 7?

**Comment by wyze** [comment]
@blimmer Are you receiving the exact same error as the initial issue or something different? We actually don't have any peer dependencies on `react-dom` only `react`. So this is coming from one of our dependencies and it may be out of our control, but can try using resolutions or something.

**Comment by blimmer** [comment]
@wyze - yep, if you make sure you have npm v7 (`npm i -g npm@7`) and try to install `@nivo/core` it fails.

```console
 npm --version
7.11.2
```

```console
cd /tmp && \
   mkdir nivo-example && \
   cd nivo-example && \
   npm init -y && \
   npm install --save @nivo/core
```

```
npm WARN ERESOLVE overriding peer dependency
npm WARN Found: react@17.0.2
npm WARN node_modules/react
npm WARN   peer react@">= 16.8.4 < 18.0.0" from @nivo/core@0.69.0
npm WARN   node_modules/@nivo/core
npm WARN     @nivo/core@"*" from the root project
npm WARN     1 more (@nivo/tooltip)
npm WARN   3 more (@nivo/recompose, @react-spring/core, @react-spring/konva)
npm WARN
npm WARN Could not resolve dependency:
npm WARN peer react@"16.8.x" from react-konva@16.8.6
npm WARN node_modules/react-konva
npm WARN   peer react-konva@">=16.8" from @react-spring/konva@9.1.2
npm WARN   node_modules/@react-spring/konva
npm WARN ERESOLVE overriding peer dependency
npm WARN Found: react@17.0.2
npm WARN node_modules/react
npm WARN   peer react@">= 16.8.4 < 18.0.0" from @nivo/core@0.69.0
npm WARN   node_modules/@nivo/core
npm WARN     @nivo/core@"*" from the root project
npm WARN     1 more (@nivo/tooltip)
npm WARN   3 more (@nivo/recompose, @react-spring/core, @react-spring/konva)
npm WARN
npm WARN Could not resolve dependency:
npm WARN peer react@"^16.0.0" from react-dom@16.8.6
npm WARN node_modules/react-dom
npm WARN   peer react-dom@"16.8.x" from react-konva@16.8.6
npm WARN   node_modules/react-konva
npm WARN ERESOLVE overriding peer dependency
npm WARN Found: konva@7.2.5
npm WARN node_modules/konva
npm WARN   peer konva@">=2.6" from @react-spring/konva@9.1.2
npm WARN   node_modules/@react-spring/konva
npm WARN     @react-spring/konva@"~9.1.2" from react-spring@9.1.2
npm WARN     node_modules/react-spring
npm WARN
npm WARN Could not resolve dependency:
npm WARN peer konva@"^3.2.3" from react-konva@16.8.6
npm WARN node_modules/react-konva
npm WARN   peer react-konva@">=16.8" from @react-spring/konva@9.1.2
npm WARN   node_modules/@react-spring/konva
npm WARN ERESOLVE overriding peer dependency
npm WARN Found: react@17.0.2
npm WARN node_modules/react
npm WARN   peer react@">= 16.8.4 < 18.0.0" from @nivo/core@0.69.0
npm WARN   node_modules/@nivo/core
npm WARN     @nivo/core@"*" from the root project
npm WARN     1 more (@nivo/tooltip)
npm WARN   3 more (@nivo/recompose, @react-spring/core, @react-spring/konva)
npm WARN
npm WARN Could not resolve dependency:
npm WARN peer react@"17.0.1" from react-native@0.64.1
npm WARN node_modules/react-native
npm WARN   peer react-native@">=0.58" from @react-spring/native@9.1.2
npm WARN   node_modules/@react-spring/native
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR!
npm ERR! Found: react@17.0.2
npm ERR! node_modules/react
npm ERR!   peer react@">= 16.8.4 < 18.0.0" from @nivo/core@0.69.0
npm ERR!   node_modules/@nivo/core
npm ERR!     @nivo/core@"*" from the root project
npm ERR!     peer @nivo/core@"0.69.0" from @nivo/tooltip@0.69.0
npm ERR!     node_modules/@nivo/tooltip
npm ERR!       peer @nivo/tooltip@"0.69.0" from @nivo/core@0.69.0
npm ERR!   peer react@">= 16.8.4 < 18.0.0" from @nivo/recompose@0.69.0
npm ERR!   node_modules/@nivo/recompose
npm ERR!     @nivo/recompose@"0.69.0" from @nivo/core@0.69.0
npm ERR!     node_modules/@nivo/core
npm ERR!       @nivo/core@"*" from the root project
npm ERR!       1 more (@nivo/tooltip)
npm ERR!   2 more (@react-spring/core, @react-spring/konva)
npm ERR!
npm ERR! Could not resolve dependency:
npm ERR! peer react@"^16.0.0" from react-reconciler@0.20.4
npm ERR! node_modules/react-konva/node_modules/react-reconciler
npm ERR!   react-reconciler@"^0.20.4" from react-konva@16.8.6
npm ERR!   node_modules/react-konva
npm ERR!     peer react-konva@">=16.8" from @react-spring/konva@9.1.2
npm ERR!     node_modules/@react-spring/konva
npm ERR!       @react-spring/konva@"~9.1.2" from react-spring@9.1.2
npm ERR!       node_modules/react-spring
npm ERR!
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR!
npm ERR! See /Users/blimmer/.npm/eresolve-report.txt for a full report.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/blimmer/.npm/_logs/2021-05-05T21_44_04_783Z-debug.log
```

**Comment by wyze** [comment]
Okay, well looks like a change might be needed in the `react-zdog` package to upgrade its `react-reconciler` version. In the meantime, you could use `--legacy-peer-deps` flag when installing to bypass the warning/errors.

**Comment by blimmer** [comment]
Btw, this appears to be fixed for me now with `0.71.1` - thanks!
